test/rng_testsuite/statistics_tests/exponential_distribution_test.pass.cpp

```diff
@@ -0,0 +1,638 @@
+                oneapi::dpl::exponential_distribution<RealType> distr(lambda);
+
+                sycl::vec<oneapi::dpl::internal::element_type_t<RealType>, num_elems> res = distr(engine);
+                res.store(idx.get_linear_id(), dpstd_acc.get_pointer());
+            });
+        });
+        queue.wait();
+    }
+
+    // statistics check
+    int err = statistics_check(nsamples, lambda, dpstd_samples);
+
+    if (err)
+    {
+        std::cout << "\tFailed" << std::endl;
+    }
+    else
+    {
+        std::cout << "\tPassed" << std::endl;
+    }
+
+    return err;
+}
+
+template <class RealType, class UIntType>
+int
+test_portion(oneapi::dpl::internal::element_type_t<RealType> lambda, int nsamples, unsigned int part)
+{
+
+    sycl::queue queue(sycl::default_selector{});
+
+    // memory allocation
+    std::vector<oneapi::dpl::internal::element_type_t<RealType>> dpstd_samples(nsamples);
+    constexpr unsigned int num_elems = oneapi::dpl::internal::type_traits_t<RealType>::num_elems == 0
+                                           ? 1
+                                           : oneapi::dpl::internal::type_traits_t<RealType>::num_elems;
+    int n_elems = (part >= num_elems) ? num_elems : part;
+
+    // dpstd generation
+    {
+        sycl::buffer<oneapi::dpl::internal::element_type_t<RealType>, 1> dpstd_buffer(dpstd_samples.data(), nsamples);
+
+        queue.submit([&](sycl::handler& cgh) {
+            auto dpstd_acc = dpstd_buffer.template get_access<sycl::access::mode::write>(cgh);
+
+            cgh.parallel_for<>(sycl::range<1>(nsamples / n_elems), [=](sycl::item<1> idx) {
+                unsigned long long offset = idx.get_linear_id() * n_elems;
+                oneapi::dpl::linear_congruential_engine<UIntType, a, c, m> engine(seed, offset);
+                oneapi::dpl::exponential_distribution<RealType> distr(lambda);
+
+                sycl::vec<oneapi::dpl::internal::element_type_t<RealType>, num_elems> res = distr(engine, part);
+                for (int i = 0; i < n_elems; ++i)
+                    dpstd_acc.get_pointer()[offset + i] = res[i];
+            });
+        });
+        queue.wait_and_throw();
+    }
+
+    // statistics check
+    int err = statistics_check(nsamples, lambda, dpstd_samples);
+
+    if (err)
+    {
+        std::cout << "\tFailed" << std::endl;
+    }
+    else
+    {
+        std::cout << "\tPassed" << std::endl;
+    }
+
+    return err;
+}
+
+template <class RealType, class UIntType>
+int
+tests_set(int nsamples)
+{
+    constexpr int nparams = 2;
+
+    float lambda_array[nparams] = {0.5, 1.5};
+
+    int err;
+    // Test for all non-zero parameters
+    for (int i = 0; i < nparams; ++i)
+    {
+        std::cout << "exponential_distribution test<type>, lambda = " << lambda_array[i]
+                  << ", nsamples  = " << nsamples;
+        err = test<RealType, UIntType>(lambda_array[i], nsamples);
+        if (err)

```

Comment:
since this is the only one place where `err` is used, I'd suggest to get rid of the variable

Reply:
done